xlators/cluster/afr/src/afr-self-heal-common.c

```diff
@@ -380,6 +380,21 @@ afr_gfid_split_brain_source(xlator_t *this, struct afr_reply *replies,
         default:
             break;
     }
+    /* At this point we have a source selected by favourite child policy,
+     * If this is for a directory it might not be a good idea to automatically
+     * resolve the GFID. Because the ultimate view of a dir is owned by DHT.
+     * But if it is not a distributed volume, then it might be fine to do
+     * the automatic gfid fix.
+     * So here we take the decesion based on a configurable option.
+     */
+    if (*src != -1 && !priv->heal_dir_gfid_mismatch &&
+        (IA_ISDIR(replies[*src].poststat.ia_type))) {
+        gf_msg(this->name, GF_LOG_INFO, 0, AFR_MSG_SPLIT_BRAIN,
+               "Automatic Gfid mismatch resolution for directories will be "
+               "skipped due to disabled option gfid-mismatch-automatic-heal. "
+               "Please manually resolve the splitbrain using cli");
+        *src = -1;
+    }
```

Comment:
@rafikc30 I think adding a new option `priv->heal_dir_gfid_mismatch` just to allow heal for plain replicate case is an overkill. Replica 2 volumes are not recommended anyway and in replica 3, gfid mismatch is rare. Let us unconditionally fail gfid split-brain resolution for directories for favorite child policy. Code will also be simpler. You can add this check just before the switch case.

Reply:
@itisravi Sure, I will make these changes. I added the option our organization uses simple replica volume with fixed quorum too. But I think this is not the default configuration that most of the community uses, So I will just remove the option and make it simple.

Reply:
@itisravi I added them after the switch case is because we will have a valid src to check whether the selected source is dir or not. Otherwise, Can I arbitrarily select the first valid successful reply from the list to check whether this is dir or not?

Reply:
@rafikc30  Can't we use inode->ia_type?  Also please change the commit message since you are not adding a new option anymore.

Reply:
@itisravi if it is the first lookup the inode won't be linked, right? If so, it will have an invalid value.

Reply:
@rafikc30 you're right, looks like it will never be linked in this code path. Let us just update the commit message.

Reply:
@itisravi I have updated the commit message

Reply:
@rafikc30  I still see "afr/heal: Introduce an option to skip gfid mismatch heal " and one more commit that corrects the `gf_msg`

Reply:
@itisravi I have corrected it

Reply:
@rafikc30 Doesn't look like it. I see 

```
afr/heal: Introduce an option to skip gfid mismatch heal
```
Maybe use something like
`afr/heal: avoid gfid heal of directories using favourite child policy` or something more approrpiate.

Reply:
I think you edited the github PR page. That doesn't change the actual commit message.